stetho/src/main/java/com/facebook/stetho/inspector/protocol/module/Page.java

```diff
@@ -9,25 +9,28 @@
 
 package com.facebook.stetho.inspector.protocol.module;
 
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.Iterator;
-import java.util.List;
-
 import android.content.Context;
 import com.facebook.stetho.common.ProcessUtil;
+import com.facebook.stetho.inspector.domstorage.SharedPreferencesHelper;
 import com.facebook.stetho.inspector.jsonrpc.JsonRpcPeer;
 import com.facebook.stetho.inspector.jsonrpc.JsonRpcResult;
 import com.facebook.stetho.inspector.protocol.ChromeDevtoolsDomain;
 import com.facebook.stetho.inspector.protocol.ChromeDevtoolsMethod;
-import com.facebook.stetho.inspector.domstorage.SharedPreferencesHelper;
+import com.facebook.stetho.json.ObjectMapper;
 import com.facebook.stetho.json.annotation.JsonProperty;
 import com.facebook.stetho.json.annotation.JsonValue;
-
+import com.facebook.stetho.screencast.android.ScreencastDispatcher;
 import org.json.JSONObject;
 
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+
 public class Page implements ChromeDevtoolsDomain {
   private final Context mContext;
+  private final ScreencastDispatcher mScreencastDispatcher = new ScreencastDispatcher();
```

Comment:
Related to my comment, 'what if multiple peers want to initiate screencasting?' ...

I'd like to recommend a design change here. First, I don't think we should construct a ScreencastDispatcher until we receive a screencasting request. When screencasting stops, we need to completely dispose of all the resources associated with screencasting (like the `HandlerThread`!), including the setting of mScreencastDispatcher to null. 

If a 2nd peer initiates screencasting, `ScreencastDispatcher` should be able to add that to its broadcast list (e.g. instead of `mPeer.invokeMethod()` we would loop over a thread-safe `List` of peers). This last part in particular might be kind of frustrating at this point, especially since I communicated that everything was looking pretty good and almost ready to merge, and also because I'm not sure that this is needed "in practice." So, I'd support a short-term strategy of simply refusing a 2nd peer's screencasting request, and then filing a new issue to support multi-peer screencasting.